lib/codebreaker/game.rb

```diff
@@ -0,0 +1,75 @@
+
+module Codebreaker
+  # //
+  class Game
+    attr_reader :attempts_left, :answer, :hints_left, :history
+
+    def initialize
+      new_game
+    end
+
+    def new_game
+      @secret = create_code
+      @history = {}
+      @answer = ''
+      @hints_left = HINTS_AMOUNT
+      @attempts_left = ATTEMPTS_AMOUNT
+    end
+
+    def answer_on(input)
+      @answer = ''
+      validate input
+      form_an_answer_for input
+      @history[input] = answer
+      @attempts_left -= 1
+      answer
+    end
+
+    def won?
+      @answer == '++++'
+    end
+
+    def hint
+      return NO_HINTS_LEFT if hints_left.zero?
+      @hints_left -= 1
+      @secret[rand(0..3)]
+    end
+
+    def used_attempts
+      ATTEMPTS_AMOUNT - attempts_left
+    end
+
+    def used_hints
+      HINTS_AMOUNT - hints_left
+    end
+
+    def game_over?
+      won? || attempts_left.zero?
+    end
+
+    private
+
+    def create_code
+      (1..4).map { rand(1..6) }.join
+    end
+
+    def form_an_answer_for(input)
+      temp = @secret.clone
```

Comment:
Please check your code for accessing and overriding instance variables and try to use ruby getters and setters for the same purpose.